### ISSUE TITLE
Documenting multiple return types with google docstrings

### ISSUE THREAD

**Issue by Crozzers** [mitmproxy/pdoc#265]
#### Problem Description

When documenting multiple return types for a function it is not parsed the same as the "Args:" section. Each new line is joined to the previous one.
```
Returns:
    int: type 1
    str: type 2
```
gets translated as
```
Returns:
> int: type 1 str: type 2
```
instead of
```
Returns:
    - **int**: type 1
    - **str**: type 2
```

#### Proposal
If the "Returns:" section contains what looks like multiple return types then parse it in a similar way to the "Args:" section.
EG: if there are multiple lines and all of them are in the format `type: description`.
Otherwise, parse it as per usual.

#### Alternatives
Since this isn't referenced anywhere in the google docstring styleguide this may be undesired default behaviour.
Therefore, instead of this becoming an in-built feature, this could be added as a configuration toggle/command line flag.
EG: `pdoc -d google --multi-line-returns test.py` or `pdoc.render.configure(docformat='google', multi_line_returns=True)`

#### Additional context

Some examples of how different types of "Returns:" sections might be parsed.

![image](https://user-images.githubusercontent.com/57498990/120088177-055e2880-c0e6-11eb-976c-0e603df47eee.png)

**Comment by mhils** [comment]
Hi there! Thanks for the suggestion. :)

> Therefore, instead of this becoming an in-built feature, this could be added as a configuration toggle/command line flag.

Irrespective of what's below: I don't want to add additional configuration toggles/command line flags for such niche functionality.

> Since this isn't referenced anywhere in the google docstring styleguide this may be undesired default behaviour.

Here's what the Google style guide says:

```
Returns: (or Yields: for generators)
    Describe the type and semantics of the return value. If the function only returns None, 
    this section is not required. It may also be omitted if the docstring starts with Returns 
    or Yields (e.g. """Returns row from Bigtable as a tuple of strings.""") and the opening 
    sentence is sufficient to describe return value.
```

To me, this reads like this section is mostly expected to be prose. In contrast, *Args* is explicitly mentioned to be a list. 

Numpydoc also describes *Returns* as having only one type:

https://github.com/mitmproxy/pdoc/blob/e59c3c6db65813770d386ff479ea9dc4ca8fcfef/test/testdata/flavors_google.py#L117-L132

On the other hand, `docstring_parser` has it of type ["singular or multiple"](https://github.com/rr-/docstring_parser/blob/76637ab2591a56383076520c0e4041b088c7735b/docstring_parser/google.py#L56).

Long story short, opinions differ here. I would be open to support both, but that comes with quite some implementation complexity, in particular once people start mixing typed lists with prose and so on. I don't think this is a rabbit hole we should get into. I feel that treating this section as prose is a reasonable tradeoff, if you absolutely need multiple return types you can just add a little bit of Markdown and you have your desired formatting. Does that make sense?

**Comment by Crozzers** [comment]
That's fair. Markdown is very human readable anyway so adding that in a docstring isn't too bad.

Perhaps, though, in `pdoc/docstring.py` in the function `_google_section`, instead of having
```python
if name in ("Args", "Raises", "Attributes"):
```
as a hard-coded tuple it could be a constant defined in that file that a user could add to if they wished?
Something like
```python
if name in MULTILINE_SECTIONS:

...

MULTILINE_SECTIONS = ["Args", "Raises", "Attributes"]
```
That way if a user really wanted they could have multi-line returns and they could also add custom sections to be parsed. For example, if a user wanted a "Class Attributes" section and an "Instance Variables" section. Then again, there's already ways to document each of these things.

**Comment by mhils** [comment]
I'd be happy to add a `GOOGLE_LIST_SECTIONS` constant if that is helpful to you. Feel free to send a PR! 😃

**Comment by Crozzers** [comment]
PR submitted. Thank you for your time
